sor/src/main/java/com/bazaarvoice/emodb/sor/db/DeltaIterator.java

```diff
@@ -128,46 +130,58 @@ protected T computeNext() {
                 return endOfData();
             }
 
-            ByteBuffer content;
+            BlockedDelta blockedDelta;
 
             if (!_reverse) {
 
+                // Ensure that the block we are looking at is truly the first block in the sequence. It not,
+                // we should skip over it.
+                // TODO: add a metric here so that we know if this is happening. The only case where this should be
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.qa.bazaarvoice.com/coding_rules#rule_key=squid%3AS1135)